src/main/java/uk/gov/justice/hmpps/prison/repository/OffenderCurfewRepository.java

```diff
@@ -69,6 +80,7 @@ public Collection<OffenderCurfew> offenderCurfews(final Set<String> agencyIds) {
 
 
     public void setHDCChecksPassed(final long curfewId, final HdcChecks hdcChecks) {
+        getCurfewLock(curfewId);
```

Comment:
Was it not possible to combine the "FOR UPDATE WAIT" in the actual SQL about to be executed? E.g change the SQL in say OffenderCurfewRepositorySql.UPDATE_CURFEW_CHECKS_PASSED ?

Reply:
Do you mean combine the select for update with the update statement in the sql string?